mypy/checkexpr.py

```diff
@@ -4727,7 +4738,14 @@ def accept(
         applies only to this expression and not any subexpressions.
         """
         if node in self.type_overrides:
+            # This branch is very fast, there is no point timing it.
             return self.type_overrides[node]
+        # We don't use context manager here to get most precise data (and avoid overhead).
+        record_time = False
+        if not self.in_expression and self.chk.options.line_checking_stats is not None:
```

Comment:
Since this method is hot, what about caching the value of `self.chk.options.line_checking_stats is not None` in `__init__`? Then this check could be written like this, for example (this first condition would almost always be false, making this quick):
```py
if self.collect_line_checking_stats and not self.in_expression:
    ...
```